config.go

```diff
@@ -0,0 +1,10 @@
+package plugin
+
+// Configor is the interface plugins should implement in order to provide configuration interface to the user
+type Configor interface {
+	Plugin
+	// DefaultConfig will be called on plugin initialization. Plugins should return a default configuration in YAML format. The default configuration will be provided to the user for future edit.
+	DefaultConfig() []byte
+	// LoadConfig will be called every time the plugin is initialized or the configuration has been changed by the user. Plugins should check whether the configuration is valid and optionally return an error.
+	LoadConfig(c []byte) error
```

Comment:
Maybe this could also rename this method to SetConfig because the config is already loaded, the plugin only must process it.

Reply:
Actually here I designed the Configor interface to be used in this way:

1. On plugin initialization, `LoadConfig` is called with the current configuration recalled from the database(or DefaultConfig() if there is none)
1. When user updated the configuration, the configuration is sent to the plugin instance first by calling `LoadConfig` before saving the changes to the database, if the plugin returned error, neither the plugin should update the configuration state not the data in the database should be updated. Only when the plugin returned nil error the configuration is considered accepted and saved to the database.

Reply:
Understood, how about having ValidateConfig(interface{}) and SetConfig(interface{}) method? this seems a little more obvious to me.

Reply:
Sounds fine